src/python/pants/util/contextutil.py

```diff
@@ -22,6 +22,13 @@
 from pants.util.tarutil import TarFile
 
 
+class InvalidZipPath(zipfile.BadZipfile):
```

Comment:
Sure, I can do that.

I was doing the song and dance to not break the public API of raising BadZipfile. But I am persuaded by your argument that it is not documented, therefore fair game :)